x-pack/plugins/apm/server/routes/services/get_service_container_metadata.ts

```diff
@@ -0,0 +1,148 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { ElasticsearchClient } from '@kbn/core/server';
+import { estypes } from '@elastic/elasticsearch';
+import { rangeQuery } from '@kbn/observability-plugin/server';
+import {
+  CONTAINER,
+  CONTAINER_ID,
+  CONTAINER_IMAGE,
+  KUBERNETES,
+  KUBERNETES_CONTAINER_NAME,
+  KUBERNETES_NAMESPACE,
+  KUBERNETES_POD_NAME,
+  KUBERNETES_POD_UID,
+  KUBERNETES_REPLICASET_NAME,
+  KUBERNETES_DEPLOYMENT_NAME,
+} from '../../../common/elasticsearch_fieldnames';
+import { Kubernetes } from '../../../typings/es_schemas/raw/fields/kubernetes';
+import { Container } from '../../../typings/es_schemas/raw/fields/container';
+import { maybe } from '../../../common/utils/maybe';
+
+export interface ContainerMetadata {
+  kubernetes?: Kubernetes;
+  container?: Container;
+}
+
+export interface ResponseHitSource {
+  [key: string]: {
+    pod: {
+      name: string;
+      uid: string;
+    };
+    image: {
+      name: string;
+    };
+    id: string;
+  };
+}
+
+export interface ResponseHit {
+  _source: ResponseHitSource;
+}
+
+interface Aggs extends estypes.AggregationsMultiBucketAggregateBase {
+  buckets: Array<{
+    key: string;
+    key_as_string?: string;
+  }>;
+}
```

Comment:
Why do we need explicit types? Is the type inference in `esClient.search` not sufficient?

Reply:
it's not sufficient for the `Aggregations`. Cleaned up the type here https://github.com/elastic/kibana/pull/139612/commits/3cf8d565807257c70eca6c9cef8efe06af6171b3